invenio_app_ils/ui/src/invenio_app_ils/common/components/AuthenticatedRoute/AuthenticatedRoute.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { sessionManager, authenticationService } from '../../../authentication';
+
+export class AuthenticatedRoute extends Component {
+  render() {
+    let { component: Component, roles, ...restParams } = this.props;
+
+    if (!sessionManager.isUserLoggedIn()) {
+      authenticationService.login(this.props.location.pathname);
+      return null;
+    }
+
+    if (
+      sessionManager.isUserLoggedIn() &&
+      !sessionManager.hasUserNeededRoles(roles)
+    ) {
+      // TODO: Handle the insufficient permissions for user
```

Comment:
Create a component for the unauthorized?
If you don't want to do it now, please issue for it :)